plugins/plugin-sass/plugin.js

```diff
@@ -0,0 +1,76 @@
+const execa = require('execa');
+const fs = require('fs');
+const path = require('path');
+
+const IMPORT_REGEX = /\@(use|import)\s*['"](.*?)['"]/g;
+
+function scanSassImports(fileContents, filePath) {
+  // TODO: Replace with matchAll once Node v10 is out of TLS.
+  // const allMatches = [...result.matchAll(new RegExp(HTML_JS_REGEX))];
+  const allMatches = [];
+  let match;
+  const regex = new RegExp(IMPORT_REGEX);
+  while ((match = regex.exec(fileContents))) {
+    allMatches.push(match);
+  }
+
+  return allMatches
+    .map((match) => match[2])
+    .filter((s) => s.trim())
+    .map((s) => {
+      if (!path.extname(s)) {
+        s += '.scss';
+      }
+      return path.resolve(path.dirname(filePath), s);
+    });
+}
+
+module.exports = function postcssPlugin(_, options) {
+  const importedByMap = new Map();
+
+  function addImportsToMap(filePath, sassImports) {
+    for (const imported of sassImports) {
+      const importedBy = importedByMap.get(imported);
+      if (importedBy) {
+        importedBy.add(filePath);
+      } else {
+        importedByMap.set(imported, new Set([filePath]));
+      }
+    }
+  }
+
+  return {
+    name: '@snowpack/plugin-sass',
+    resolve: {
+      input: ['.scss'],
+      output: ['.css'],
+    },
+    /** when a file changes, also mark it's importers as changed. */
+    onChange({filePath}) {
+      const importedBy = importedByMap.get(filePath);
+      if (!importedBy) {
+        return;
+      }
+      importedByMap.delete(filePath);
+      for (const importerFilePath of importedBy) {
+        this.markChange(importerFilePath);
+      }
+    },
+    /** Load the Sass file and compile it to CSS. */
+    async load({filePath, isDev}) {
+      const contents = fs.readFileSync(filePath, 'utf8');
+      // During development, we need to track changes to Sass dependencies.
+      if (isDev) {
+        const sassImports = scanSassImports(contents, filePath);
+        addImportsToMap(filePath, sassImports);
+      }
+      // Build the file.
+      const {stdout, stderr} = await execa('sass', ['--stdin', '--load-path', path.dirname(filePath)], {
```

Comment:
👍🏻 I’ve been thinking about this plugin for a while, and I agree in using the CLI version.

I’ve also been wondering if we could use the [JS wrapper around Dart Sass](https://sass-lang.com/dart-sass):

> The pure JS version is slower than the stand-alone executable, but it's easy to integrate into existing workflows and it allows you to define custom functions and importers in JavaScript.

Do you think there’s anything that the JS wrapper can get us here?

Reply:
See: https://github.com/pikapkg/snowpack/pull/1225#discussion_r499940623

Reply:
Right—understood about configuration, but I was trying to ask something different here. I was wondering if we could replace some of our custom import logic with some method provided by the JS wrapper, but after [skimming the docs](https://github.com/sass/dart-sass/blob/master/README.md#javascript-api), nothing jumps out as solving that problem. Just wanted to raise it in case you noticed something that could save some time/management later.

Reply:
ah I missed your comment. Yes! Added this support to the package, and adding docs to the README